File: tests/machine/x/pl/match_full.pl
:- style_check(-singleton).
classify(N, _Res) :-
    (N == 1 -> _V0 = "one" ; _V0 = "many"),
    (N == 0 -> _V1 = "zero" ; _V1 = _V0),
    _Res = _V1.

:- initialization(main, main).
main :-
    X is 2,
    (X == 3 -> _V0 = "three" ; _V0 = "unknown"),
    (X == 2 -> _V1 = "two" ; _V1 = _V0),
    (X == 1 -> _V2 = "one" ; _V2 = _V1),
    Label = _V2,
    writeln(Label),
    Day = "sun",
    (Day == "sun" -> _V3 = "relaxed" ; _V3 = "normal"),
    (Day == "fri" -> _V4 = "excited" ; _V4 = _V3),
    (Day == "mon" -> _V5 = "tired" ; _V5 = _V4),
    Mood = _V5,
    writeln(Mood),
    Ok = true,
    (Ok == true -> _V6 = "confirmed" ; _V6 = "denied"),
    Status = _V6,
    writeln(Status),
    classify(0, _V7),
    writeln(_V7),
    classify(5, _V8),
    writeln(_V8),
    true.

File: tests/human/x/pas/match_full.pas
program MatchFull;

function classify(n: integer): string;
begin
  case n of
    0: classify := 'zero';
    1: classify := 'one';
  else
    classify := 'many';
  end;
end;

var
  x: integer;
  day: string;
  mood: string;
  ok: boolean;
  labelStr: string;
begin
  x := 2;
  case x of
    1: labelStr := 'one';
    2: labelStr := 'two';
    3: labelStr := 'three';
  else
    labelStr := 'unknown';
  end;
  Writeln(labelStr);

  day := 'sun';
  if day = 'mon' then mood := 'tired'
  else if day = 'fri' then mood := 'excited'
  else if day = 'sun' then mood := 'relaxed'
  else mood := 'normal';
  Writeln(mood);

  ok := true;
  if ok then labelStr := 'confirmed' else labelStr := 'denied';
  Writeln(labelStr);

  Writeln(classify(0));
  Writeln(classify(5));
end.
